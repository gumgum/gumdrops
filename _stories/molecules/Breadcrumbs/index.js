import React, { Component } from 'react';
import { select, boolean } from '@storybook/addon-knobs';

import readme from './README.md';
import Breadcrumbs from '../../../components/molecules/Breadcrumbs';

const optionsA = [
    '/',
    '/home',
    '/home/static-page',
    '/home/category',
    '/home/category/subcategory-1',
    '/home/category/subcategory-2',
    '/home/category/subcategory-3',
    '/home/products/42',
    '/home/products/42/edit',
    '/home/products/stringId',
    '/home/products/stringId/edit'
];

const optionsB = [
    '/publishers',
    '/publishers/42',
    '/publishers/42/settings',
    '/publishers/42/financials',
    '/publishers/42/zones',
    '/publishers/42/performance',
    '/publishers/42/users',
    '/publishers/42/activity-log',
    '/zones',
    '/zones/woop',
    '/zones/woop/overview',
    '/zones/woop/overview/settings'
];

const configA = {
    title: 'Home',
    path: 'home',
    subpaths: [
        {
            title: 'Main Category',
            path: 'category',
            subpaths: [
                {
                    title: 'Sub Category 1',
                    path: 'subcategory-1'
                },
                {
                    title: 'Sub Category 2',
                    path: 'subcategory-2'
                },
                {
                    title: 'Sub Category 3',
                    path: 'subcategory-3'
                }
            ]
        },
        {
            title: 'Products',
            path: 'products',
            subpaths: [
                {
                    path: ':id',
                    subpaths: [
                        {
                            path: 'edit'
                        }
                    ]
                }
            ]
        },
        {
            path: ':pageId'
        }
    ]
};

const configB = {
    title: 'Start',
    path: '/',
    subpaths: [
        {
            path: 'publishers',
            subpaths: [
                {
                    path: ':id',
                    subpaths: [
                        {
                            path: 'settings'
                        },
                        {
                            path: 'financials'
                        },
                        {
                            path: 'zones'
                        },
                        {
                            path: 'performance'
                        },
                        {
                            path: 'users'
                        },
                        {
                            path: 'activity-log',
                            title: 'Activity Log'
                        }
                    ]
                }
            ]
        },
        {
            path: 'zones',
            subpaths: [
                {
                    path: ':id'
                }
            ]
        }
    ]
};

const configSelect = {
    A: 'A',
    B: 'B',
    C: 'C'
};

const configs = {
    A: configA,
    B: configB,
    C: undefined
};

const options = {
    A: optionsA,
    B: optionsB,
    C: [...optionsA, ...optionsB]
};

class BreadcrumbsStory extends Component {
    static displayName = 'Breadcrumbs';

    printCode = code => (code ? '\n' + JSON.stringify(code, null, 4) + '\n\n' : '');

    render() {
        const selectedConfig = select('Configuration', configSelect, configSelect.A);

        const noConfig = selectedConfig === 'C';

        const configTitle = `//${noConfig ? ' No' : ''} Configuration${
            noConfig ? '' : ' ' + (selectedConfig + ':')
        }`;

        const config = configs[selectedConfig];

        return (
            <div>
                <header className="gds-page-header -color-bg-white">
                    <div className="gds-page-header__nav-bar">
                        <Breadcrumbs
                            config={config}
                            pathname={select(
                                'Pathname',
                                options[selectedConfig],
                                options[selectedConfig][0]
                            )}
                            hideMenus={boolean('Hide submenus', false)}
                            hideRoot={boolean('Hide root breadcrumb', false)}
                        />
                    </div>
                </header>
                <pre className="-m-t-6">
                    {configTitle}
                    {this.printCode(configs[selectedConfig])}
                </pre>
            </div>
        );
    }
}

const component = () => <BreadcrumbsStory />; // eslint-disable-line

export default [readme, component];
