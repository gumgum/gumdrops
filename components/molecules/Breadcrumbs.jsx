import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Breadcrumb from './Breadcrumb';
import BreadcrumbLink from './BreadcrumbLink';
import BreadcrumbsWrapper from './BreadcrumbsWrapper';

class Breadcrumbs extends Component {
    static displayName = 'Breadcrumbs';

    static defaultProps = {
        config: { path: '/' },
        linkComponent: BreadcrumbLink,
        titleDecorator: title =>
            title.replace(/^\w/, chr => chr.toUpperCase()).replace(/-|_/g, ' '),
        hideMenus: false,
        hideRoot: false
    };

    static propTypes = {
        linkComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]).isRequired,
        config: PropTypes.shape({
            title: PropTypes.string,
            path: PropTypes.string,
            subpaths: PropTypes.array
        }).isRequired,
        pathname: PropTypes.string.isRequired,
        titleDecorator: PropTypes.func,
        hideMenus: PropTypes.bool,
        hideRoot: PropTypes.bool
    };

    // Create the full path for the current slug based on the previous entry
    _buildHref = (arr, slug) => {
        const previous = arr && arr[arr.length - 1];
        const newPath = previous ? `${previous.path}/${slug}` : slug;
        return newPath.replace('//', '/');
    };

    // Create a title for parameters, either the capitalized param or a number
    _getParamTitle = param =>
        Number.isInteger(parseInt(param)) ? param : this.props.titleDecorator(param);

    // Creates a flat Array only with the nodes that correspond to the current pathname
    // It includes replaces the path with the full path to each link
    _findTrail = (pathname, config) => {
        const initialBreadcrumb = {
            title: config.title || '/',
            path: config.path.charAt(0) === '/' ? config.path : `/${config.path || ''}`
        };

        const breadcrumbConfig = {
            title: config.title || '/',
            path: config.path
        };

        // Set initial data if necessary
        const initialData = config.path === '/' ? [initialBreadcrumb] : [breadcrumbConfig];

        const searchBreakpoints = (pathSections, subpaths, accumulator = []) =>
            pathSections.reduce((trail, pathSection, index) => {
                // Find current path subpathData
                const subpathData = subpaths && subpaths.find(({ path }) => path === pathSection);
                // Find section param subpathData if no exact path was found
                const paramData =
                    !subpathData && subpaths && subpaths.find(({ path }) => path.charAt(0) === ':');
                const breadcrumbData = subpathData || paramData || { path: pathSection };
                const acceptBreadcrumb = trail.length < maxLength;
                // Push section subpathData  to accumulator
                if (acceptBreadcrumb && breadcrumbData) {
                    // Set defaults depending on type of path (param or regular path)
                    const breadcrumbPath = paramData ? pathSection : breadcrumbData.path;
                    const breadcrumbTitle =
                        breadcrumbData.title || this._getParamTitle(pathSection);
                    // Build full path for link
                    const currentPath = this._buildHref(trail, breadcrumbPath);
                    trail.push({
                        ...breadcrumbData,
                        path: currentPath,
                        title: breadcrumbTitle
                    });
                }
                const trailContinues = trail.length < maxLength;
                // Go deeper if there are subpaths
                if (breadcrumbData && breadcrumbData.subpaths && trailContinues) {
                    const nextIndex = index + 1;
                    const remainingParts = pathSections.slice(nextIndex);
                    // Search for the remaining pathname parts on the current path's subpaths,
                    return searchBreakpoints(remainingParts, breadcrumbData.subpaths, trail);
                }
                return trail;
            }, accumulator);

        // Divide pathname into sections
        const parts = pathname.split('/').filter(Boolean);
        const duplicateRoot = parts[0] === config.path;
        const initialSections = duplicateRoot ? parts.slice(1) : parts;
        const maxLength = initialSections.length + 1;
        // Compare pathname parts against user provided configuration
        const breadcrumbs = searchBreakpoints(initialSections, config.subpaths, initialData);
        return breadcrumbs;
    };

    render() {
        const { config, linkComponent, hideRoot, hideMenus, pathname } = this.props;
        const breadcrumbs = this._findTrail(pathname, config);
        const displayBreadcrumbs =
            hideRoot && breadcrumbs.length > 1 ? breadcrumbs.slice(1) : breadcrumbs;
        return (
            <BreadcrumbsWrapper>
                {displayBreadcrumbs.map(({ title, path, subpaths }, index, arr) => (
                    <Breadcrumb
                        key={path}
                        hideMenus={hideMenus}
                        linkComponent={linkComponent}
                        title={title}
                        path={path}
                        subpaths={subpaths}
                        pathname={pathname}
                        isLast={arr.length - 1 === index}
                    />
                ))}
            </BreadcrumbsWrapper>
        );
    }
}

export default Breadcrumbs;
