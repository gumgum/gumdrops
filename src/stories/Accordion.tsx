import React from 'react';
import cx from 'classnames';

interface AccordionCommonProps {
    color?: 'dark' | 'white';
    size?: 'sm' | '';
}

type OmitColor<T> = Omit<T, 'color'>;

export interface AccordionItemProps
    extends AccordionCommonProps,
        OmitColor<React.LiHTMLAttributes<HTMLLIElement>> {
    isOpen?: boolean;
    isLocked?: boolean;
    label: React.ReactChild;
}

export interface AccordionProps
    extends AccordionCommonProps,
        OmitColor<React.HTMLAttributes<Element>> {
    initialAllOpen?: boolean;
}

export const Accordion: React.FC<AccordionProps> = ({
    children,
    color,
    size = '',
    className,
    initialAllOpen,
    ...otherProps
}) => {
    const rootClass = cx('gds-accordion', className, {
        [`gds-accordion--${color}`]: color
    });

    const newChildren = React.Children.map(children, child => {
        if (!React.isValidElement<AccordionItemProps>(child)) {
            return child;
        }

        return React.cloneElement<AccordionItemProps>(child, {
            color,
            size,
            isOpen: child.props.isOpen ? child.props.isOpen : initialAllOpen
        });
    });

    return (
        <div className={rootClass} {...otherProps}>
            <ul className="gds-accordion-list">{newChildren}</ul>
        </div>
    );
};
