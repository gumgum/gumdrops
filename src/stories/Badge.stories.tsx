import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Badge, BadgeProps } from './Badge';
import { Colors } from '../types';

export default {
    title: 'Badge',
    component: Badge
} as Meta;

const Template: Story<BadgeProps> = args => <Badge {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    color: Colors.primary,
    text: 'Primary badge'
};

export const Warning = Template.bind({});
Warning.args = {
    color: Colors.warning,
    text: 'Warning Badge'
};

export const Empty = Template.bind({});
Empty.args = {
    color: Colors.info,
    empty: true
};
