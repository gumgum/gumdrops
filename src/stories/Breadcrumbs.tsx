import React from 'react';

import { BreadCrumb, Breadcrumb, LinkComponent } from './Breadcrumb';
import { BreadcrumbLink } from './BreadcrumbLink';
import { BreadcrumbsWrapper } from './BreadcrumbsWrapper';

export interface BreadcrumbsProps extends React.HTMLAttributes<Element> {
    linkComponent?: LinkComponent;
    config?: BreadCrumb | BreadCrumb[];
    pathname: string;
    titleDecorator?: (title: string) => string;
    hideMenus?: boolean;
    hideRoot?: boolean;
}

const defaultTitleDecorator = (title: string): string =>
    title.replace(/^\w/, chr => chr.toUpperCase()).replace(/-|_/g, ' ');

export const Breadcrumbs: React.FC<BreadcrumbsProps> = ({
    linkComponent = BreadcrumbLink,
    pathname,
    config = { title: null, path: '/' },
    titleDecorator = defaultTitleDecorator,
    hideMenus = false,
    hideRoot = false,
    ...otherProps
}) => {
    // Create the full path for the current slug based on the previous entry
    const _buildHref = (arr: BreadCrumb[], slug: string): string => {
        const previous = arr && arr[arr.length - 1];
        const newPath = previous ? `${previous.path}/${slug}` : slug;
        return newPath.replace('//', '/');
    };

    // Create a title for parameters, either the capitalized param or a number
    const _getParamTitle = (param: string): string =>
        Number.isInteger(parseInt(param)) ? `${param}` : titleDecorator(param as string);

    // Creates a flat Array only with the nodes that correspond to the current pathname
    // It includes replaces the path with the full path to each link
    const _findTrail = (pathname: string, config: BreadCrumb): BreadCrumb[] => {
        const initialBreadcrumb: BreadCrumb = {
            title: config.title || '/',
            path: config.path?.charAt(0) === '/' ? config.path : `/${config.path || ''}`,
            label: config.label
        };

        const breadcrumbConfig: BreadCrumb = {
            title: config.title || '/',
            path: config.path || '/',
            label: config.label
        };

        // Set initial data if necessary
        const initialData: BreadCrumb[] =
            config.path === '/' ? [initialBreadcrumb] : [breadcrumbConfig];

        const searchBreakpoints = (
            pathSections: string[],
            subpaths: BreadCrumb[],
            accumulator: BreadCrumb[] = []
        ): BreadCrumb[] =>
            pathSections.reduce((trail, pathSection, index) => {
                // Find current path subpathData
                const subpathData = subpaths && subpaths.find(({ path }) => path === pathSection);
                // Find section param subpathData if no exact path was found
                const paramData =
                    !subpathData && subpaths && subpaths.find(({ path }) => path.charAt(0) === ':');
                const breadcrumbData = subpathData || paramData || { path: pathSection };
                const acceptBreadcrumb = trail.length < maxLength;
                // Push section subpathData  to accumulator
                if (acceptBreadcrumb && breadcrumbData) {
                    // Set defaults depending on type of path (param or regular path)
                    const breadcrumbPath = paramData ? pathSection : breadcrumbData.path;
                    const breadcrumbTitle = breadcrumbData.title || _getParamTitle(pathSection);
                    // Build full path for link
                    const currentPath = _buildHref(trail, breadcrumbPath);

                    const fallbackLabel =
                        typeof breadcrumbTitle === 'string'
                            ? (breadcrumbData.title as string)
                            : currentPath;
                    const label = breadcrumbData.label;

                    const bc = {
                        ...breadcrumbData,
                        path: currentPath,
                        title: breadcrumbTitle,
                        label: label || fallbackLabel
                    };
                    trail.push(bc);
                }
                const trailContinues = trail.length < maxLength;
                // Go deeper if there are subpaths
                if (breadcrumbData && breadcrumbData.subpaths && trailContinues) {
                    const nextIndex = index + 1;
                    const remainingParts = pathSections.slice(nextIndex);
                    // Search for the remaining pathname parts on the current path's subpaths,
                    return searchBreakpoints(remainingParts, breadcrumbData.subpaths, trail);
                }
                return trail;
            }, accumulator);

        // Divide pathname into sections
        const parts = pathname.split('/').filter(Boolean);
        const duplicateRoot = parts[0] === config.path;
        const initialSections = duplicateRoot ? parts.slice(1) : parts;
        const maxLength = initialSections.length + 1;
        // Compare pathname parts against user provided configuration
        const breadcrumbs = searchBreakpoints(initialSections, config.subpaths, initialData);
        return breadcrumbs;
    };

    const breadcrumbs = Array.isArray(config) ? config : _findTrail(pathname, config);
    const displayBreadcrumbs =
        hideRoot && breadcrumbs.length > 1 ? breadcrumbs.slice(1) : breadcrumbs;
    return (
        <BreadcrumbsWrapper {...otherProps}>
            {displayBreadcrumbs.map(({ title, label, path, subpaths }, index, arr) => (
                <Breadcrumb
                    key={path}
                    hideMenus={hideMenus}
                    linkComponent={linkComponent}
                    title={title}
                    label={label}
                    path={path}
                    subpaths={subpaths}
                    pathname={pathname}
                    isLast={arr.length - 1 === index}
                />
            ))}
        </BreadcrumbsWrapper>
    );
};
