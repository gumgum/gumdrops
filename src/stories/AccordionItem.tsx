import React, { MouseEvent, KeyboardEvent, useState } from 'react';
import cx from 'classnames';
import { AccordionItemProps } from './Accordion';
import { keyCodes } from 'types';

type ToggleEvent = MouseEvent<HTMLLIElement> | KeyboardEvent<HTMLLIElement>;

export const AccordionItem: React.FC<AccordionItemProps> = ({
    isOpen: initialIsOpen,
    isLocked,
    label,
    size,
    color,
    className,
    children
}) => {
    const [isOpen, setIsOpen] = useState(initialIsOpen);

    const toggleOpen = (event: ToggleEvent): void => {
        event.stopPropagation(); // don't want nested accordions to propagate events
        event.preventDefault(); // prevents page scroll from space key
        const { type } = event;
        const { key } = event as KeyboardEvent;

        if (isLocked) return;
        const allowToggle =
            (type === 'keypress' && [keyCodes.space, keyCodes.enter].includes(key as keyCodes)) ||
            type === 'click';

        if (!allowToggle) return;

        setIsOpen(isOpen => !isOpen);
    };

    const rootClass = cx('gds-accordion__item', className, {
        [`gds-accordion__item--${color}`]: color,
        'gds-accordion__item--active': isOpen
    });

    const titleClass = cx('gds-accordion__item-title', {
        [`gds-accordion__item-title--${size}`]: size
    });

    const iconClass = cx('gds-accordion__item-icon', {
        [`gds-accordion__item-icon--${size}`]: size
    });

    const childClass = cx('gds-accordion__child-items', {
        [`gds-accordion__child-items--${size}`]: size
    });

    const newChildren = React.Children.map(children, child => {
        if (!React.isValidElement<AccordionItemProps>(child)) {
            return child;
        }

        return React.cloneElement<AccordionItemProps>(child, {
            color,
            size
        });
    });

    const regionId = `AccordionItem-region-${label}`;
    const labelId = `AccordionItem-label-${label}`;

    return (
        <li
            aria-pressed={isOpen}
            aria-controls={regionId}
            aria-expanded={isOpen}
            className={rootClass}
            onClick={toggleOpen}
            onKeyPress={toggleOpen}
            role="button"
            tabIndex={isLocked ? -1 : 0}>
            <h4 id={labelId} className={titleClass}>
                {label}
            </h4>
            {!isLocked && <span className={iconClass} />}
            <ul
                aria-labelledby={labelId}
                aria-hidden={!isOpen}
                className={childClass}
                id={regionId}
                role="region">
                {newChildren}
            </ul>
        </li>
    );
};
