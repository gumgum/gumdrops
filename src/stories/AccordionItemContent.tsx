import React from 'react';
import cx from 'classnames';
import { AccordionItemProps } from './Accordion';

export const AccordionItemContent: React.FC<Omit<AccordionItemProps, 'label'>> = ({
    className,
    color,
    size,
    children,
    ...otherProps
}) => {
    const stopClick: React.MouseEventHandler = e => {
        e.stopPropagation();
    };

    const rootClass = cx('gds-accordion__child-item', className, {
        [`gds-accordion__child-item--${color}`]: color
    });

    const titleClass = cx('gds-accordion__child-item-title', {
        [`gds-accordion__child-item-title--${size}`]: size
    });

    return (
        <li className={rootClass} onClick={stopClick} {...otherProps}>
            <h4 className={titleClass}>{children}</h4>
        </li>
    );
};
