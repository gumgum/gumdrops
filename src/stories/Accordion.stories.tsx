import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Accordion, AccordionProps } from './Accordion';
import { AccordionItem } from './AccordionItem';
import { AccordionItemContent } from './AccordionItemContent';

export default {
    title: 'Accordion',
    component: Accordion
} as Meta;

const Template: Story<AccordionProps> = args => (
    <Accordion {...args}>
        <AccordionItem label="I'm locked and open" isLocked isOpen>
            <AccordionItemContent>Locked Content</AccordionItemContent>
        </AccordionItem>
        <AccordionItem label="Item 1">
            <AccordionItemContent>Content 1</AccordionItemContent>
        </AccordionItem>
        <AccordionItem label="Item 2">
            <AccordionItemContent>Content 2</AccordionItemContent>
        </AccordionItem>
        <AccordionItem label="Item 3">
            <AccordionItemContent>Content 3</AccordionItemContent>
            <Accordion>
                <AccordionItem label="Nested Item">
                    <AccordionItemContent>Nested Content</AccordionItemContent>
                </AccordionItem>
            </Accordion>
        </AccordionItem>
    </Accordion>
);

export const Primary = Template.bind({});
Primary.args = {};
