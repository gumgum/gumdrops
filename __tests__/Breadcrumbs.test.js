import React from 'react';
import { render, screen, within } from '@testing-library/react';
import user from '@testing-library/user-event';
import { Breadcrumbs } from 'stories/Breadcrumbs';

const defaultProps = {
    className: 'cool-crumbs',
    pathname: '/home/category/subcategory-2',
    config: {
        title: 'Start',
        path: '/',
        subpaths: [
            {
                title: 'Home',
                path: 'home',
                subpaths: [
                    {
                        title: 'Main Category',
                        path: 'category',
                        subpaths: [
                            {
                                title: 'Sub Category 1',
                                path: 'subcategory-1'
                            },
                            {
                                title: 'Sub Category 2',
                                path: 'subcategory-2'
                            },
                            {
                                title: 'Sub Category 3',
                                path: 'subcategory-3'
                            }
                        ]
                    }
                ]
            }
        ]
    },
    hideMenus: false,
    hideRoot: false
};

const arrayConfig = [
    {
        title: 'Home',
        path: '/'
    },
    {
        title: 'Folder',
        path: '/folder'
    },
    {
        title: 'File',
        path: '/folder/file'
    }
];

describe('Breadcrumbs', () => {
    it('renders', () => {
        const { container } = render(<Breadcrumbs {...defaultProps} />);
        screen.getByRole('navigation', { name: 'Breadcrumbs' });
        const items = screen.getAllByRole('listitem');
        expect(within(items[0]).getByText('Start')).toHaveAttribute('href', '/');
        expect(within(items[1]).getByText('Home')).toHaveAttribute('href', '/home');
        expect(within(items[2]).getByText('Main Category')).toHaveAttribute(
            'href',
            '/home/category'
        );
        expect(within(items[3]).getByText('Sub Category 2')).not.toHaveAttribute('href');
        expect(screen.queryByText('Sub Category 1')).toBe(null);
        expect(screen.queryByText('Sub Category 3')).toBe(null);
        user.click(screen.getByRole('button', { name: 'Open Main Category menu' }));
        expect(container.firstChild).toMatchSnapshot();
        screen.getByText('Sub Category 1');
        screen.getByText('Sub Category 3');
        user.click(screen.getByRole('button', { name: 'Close Main Category menu' }));
        expect(screen.queryByText('Sub Category 1')).toBe(null);
        expect(screen.queryByText('Sub Category 3')).toBe(null);
    });

    it('renders an array config explicitly', () => {
        render(<Breadcrumbs config={arrayConfig} pathname="/folder/file" />);
        const items = screen.getAllByRole('listitem');
        expect(within(items[0]).getByText('Home')).toHaveAttribute('href', '/');
        expect(within(items[1]).getByText('Folder')).toHaveAttribute('href', '/folder');
        expect(within(items[2]).getByText('File')).not.toHaveAttribute('href');
    });
});
