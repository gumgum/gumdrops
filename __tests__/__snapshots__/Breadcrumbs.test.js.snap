// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders 1`] = `
<nav
  aria-label="Breadcrumbs"
  class="cool-crumbs"
>
  <ol
    class="gds-page-header__breadcrumbs"
  >
    <li
      aria-label="Start"
      class="gds-page-header__breadcrumbs-list-item"
    >
      <a
        class="gds-page-header__breadcrumbs-link"
        href="/"
      >
        Start
      </a>
    </li>
    <li
      aria-label="Home"
      class="gds-page-header__breadcrumbs-list-item"
    >
      <a
        class="gds-page-header__breadcrumbs-link"
        href="/home"
      >
        <i
          class="fa fa-home"
        />
         Home
      </a>
    </li>
    <li
      aria-label="Main Category"
      class="gds-page-header__breadcrumbs-list-item gds-page-header__breadcrumbs-list-item--has-menu"
    >
      <div
        aria-expanded="true"
        aria-haspopup="listbox"
        class="gds-page-header__breadcrumbs-menu"
      >
        <button
          aria-label="Close Main Category menu"
          class="gds-page-header__breadcrumbs-menu-dots"
          style="outline: none; padding: 0px;"
          type="button"
        />
        <div
          class="gds-bubble__menu gds-bubble__menu--left gds-bubble__menu--sm gds-bubble__menu--menu-open"
        >
          <ul
            aria-label="Section Menu"
            class="gds-bubble__menu-list"
          >
            <li
              class="gds-bubble__menu-list-item -ellipsis"
            >
              <a
                class="gds-bubble__menu-list-link -text-tr-cap"
                href="/home/category/subcategory-1"
              >
                Sub Category 1
              </a>
            </li>
            <li
              class="gds-bubble__menu-list-item -ellipsis"
            >
              <a
                class="gds-bubble__menu-list-link -text-tr-cap"
                href="/home/category/subcategory-3"
              >
                Sub Category 3
              </a>
            </li>
          </ul>
        </div>
      </div>
      <a
        class="gds-page-header__breadcrumbs-link"
        href="/home/category"
      >
        Main Category
      </a>
    </li>
    <li
      aria-label="Sub Category 2"
      class="gds-page-header__breadcrumbs-list-item"
    >
      Sub Category 2
    </li>
  </ol>
</nav>
`;
